internal/bgp/frr/config.go

```diff
@@ -66,21 +69,24 @@ router bgp {{.MyASN}}
   {{- end }}
 {{- if ne .BFDProfile ""}} 
   neighbor {{.Addr}} bfd profile {{.BFDProfile}}
-{{- end -}}
+{{- end }}
 {{- end }}
 {{range $n := .Neighbors -}}
 {{/* no bgp default ipv4-unicast prevents peering if no address families are defined. We declare an ipv4 one for the peer to make the pairing happen */}}
 {{- if eq (len .Advertisements) 0}}
   address-family ipv4 unicast
     neighbor {{$n.Addr}} activate
+    neighbor {{.Addr}} prefix-list ipv4deny-all in
```

Comment:
I'd recommend using route-map for in/out route filtering.

```
route-map {{.Addr}}-in permit {{.index}}
  match ip address prefix-list ipv4deny-all
route-map {{.Addr}}-in permit {{.index}}
  match ipv6 address prefix-list ipv6deny-all
...
neighbor {{.Addr}} route-map {{.Addr}}-in in
neighbor {{.Addr}} route-map {{.Addr}}-out out
```

Just will be more flexible in the future.

Reply:
will do, thanks!

Reply:
@ton31337 can FRR handle multiple route-map applied in the same direction or they have to be merged into single route-map ?

Reply:
I am not sure this version looks more readable, I may revert and bring in route-map(s) when we need to deal with more complex scenarios.
Thoughts?

Reply:
fortunately the existing route-map in the "out" direction so we don't have to merge yet :) so I prefer using route-map I understand the template becoming hard to read but I know from day1 this where it will go I will revive some initial attempt to make it a hierachy of templates not only one as it won't scale with bgp configs

Reply:
Let's try to wrap up.
We can have only one route-map per family per direction per neighbor? 
Like here it's allowed having two:

```
route-map {{.Addr}}-in permit {{.index}}
  match ip address prefix-list ipv4deny-all
route-map {{.Addr}}-in permit {{.index}}
  match ipv6 address prefix-list ipv6deny-all
``` 
because they are for different ip families?

And scaling aside, what's the reason for going route-maps now and not later (as this will always be block every route in)?

Reply:
It was just a suggestion :) 

>as this will always be block every route in

What do you mean? Which example?

Reply:
Was trying to "talk out loudly". What I meant is, what we are trying to achieve here is to always block incoming routes. Due to the nature of metallb, that is really unlikely to change so it's not that bad pinning it to a neighbor (I think).